doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+
++--------------------+-----------------+-------------------------------------------------+
+| Code               | Meaning         | Class                                           |
++====================+=================+=================================================+
+| ``(f + g)(x)``     | ``f(x) + g(x)`` | `FunctionalSum`                                 |
+|                    |                 | - Retains `gradient`.                           |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f + a)(x)``     | ``f(x) + a``    | `FunctionalScalarSum`                           |
+|                    |                 | - Retains all properties.                       |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * A)(x)``     | ``f(A(x))``     | `FunctionalComp`                                |
+|                    |                 | - Retains gradient                              |
++--------------------+-----------------+-------------------------------------------------+
+| ``(a * f)(x)``     | ``a * f(x)``    | `FunctionalLeftScalarMult`                      |
+|                    |                 | - Retains all properties, if ``a`` is positive. |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * a)(x)``     | ``f(a * x)``    | `FunctionalRightScalarMult`                     |
+|                    |                 | - Retains all properties                        |
++--------------------+-----------------+-------------------------------------------------+
+| ``(v * f)(x)``     | ``v * f(x)``    | `FunctionalLeftVectorMult`                      |
+|                    |                 | - This is not a functional, but an operator     |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * v)(x)``     | ``f(v * x)``    | `FunctionalRightVectorMult`                     |
+|                    |                 | - Retains gradient and convex conjugate.        |
++--------------------+-----------------+-------------------------------------------------+
+| ``f.translate(y)`` | ``f(. - y)``    | `TranslatedFunctional`                          |
+|                    |                 | - Retains all properties.                       |
++--------------------+-----------------+-------------------------------------------------+
+
+
+Code example
+============
+This section contains an example of an implementation of a functional, namely the functional :math:`\|x\|_2^2 + \langle x, y \rangle`.
+Another example can be found `functional_basic_example.py`, and more implementations of other functionals can be found in `default_functionals.py`. ::
```

Comment:
Instead of pasting the code, you can directly include a code file literally:

```
.. literalinclude:: code/functional_indepth_example.py
   :language: python
```